python/tests/test_pyvw.py

```diff
@@ -183,3 +183,15 @@ def test_parse():
     ex = model.parse(["| a:1 b:0.5", "0:0.1:0.75 | a:0.5 b:1 c:2"])
     assert len(ex) == 2
     del model
+
+def test_numspace_id():
+    vw_ex = vw(quiet=True)
+    ex = vw_ex.example('1 |a two features |b more features here')
+    nm1 = pyvw.namespace_id(ex, 2)
+    assert nm1.id == 2
+    assert nm1.ord_ns == 128
+    assert nm1.ns == '\x80'
+    nm2 = pyvw.namespace_id(ex, '')
```

Comment:
How does this obtain the default namespace? Seems odd that an empty string works here?

Reply:
Actually this too works here:
`nm2 = pyvw.namespace_id(ex, 'some_id')`
Should an empty string raise error for example passed here?

Reply:
Strings are valid, if you pass an empty string it is converted to a space. A space corresponds to the default namespace (32). The example you've provided doesn't included doesn't have any default namespaces.

Reply:
```
>>> nm2 = pyvw.namespace_id(ex, 'x')
assert nm2.id is None
assert nm2.ord_ns == 120
assert nm2.ns == 'x'

```
Adding this would be good?

Reply:
```
>>> nm2 = pyvw.namespace_id(ex, 'x')
assert nm2.ns == 'x'
assert nm2.ord_ns == 120
assert nm2.id is None
```
Would adding this test be good?

Reply:
You should read the implementation of `namespace_id` to get a better idea of what it does and how you should test it here

Reply:
I have corrected the tests according to the docstring of namespace_id